model.py

```diff
@@ -5471,6 +5471,47 @@ def is_open_access(self):
         return (self.rights_status
                 and self.rights_status.uri in RightsStatus.OPEN_ACCESS)
 
+    def compatible_with(self, other):
+        """Can a single loan be fulfilled with both this
+        LicensePoolDeliveryMechanism and the given one?
+
+        :param other: A LicensePoolDeliveryMechanism.
+        """
+        if not isinstance(other, LicensePoolDeliveryMechanism):
+            return False
+
+        if other.id==self.id:
+            # They two LicensePoolDeliveryMechanisms are the same object.
+            return True
+
+        # The two LicensePoolDeliveryMechanisms must be different ways
+        # of getting the same book from the same source.
+        if other.identifier_id != self.identifier_id:
+            return False
+        if other.data_source_id != self.data_source_id:
+            return False
+
+        if other.delivery_mechanism_id == self.delivery_mechanism_id:
+            # We have two LicensePoolDeliveryMechanisms for the same
+            # underlying delivery mechanism. This can happen when an
+            # open-access book gets its content mirrored to two
+            # different places.
+            return True
+
+        # If the DeliveryMechanisms themselves are compatible, then the
+        # LicensePoolDeliveryMechanisms are compatible.
+        #
+        # In practice, this means that either the two
+        # DeliveryMechanisms are the same or that one of them is a
+        # streaming mechanism.
+        open_access_rules = self.is_open_access and other.is_open_access
```

Comment:
I feel like this could be `or`? At least if the locked in mechanism is not open access, other mechanisms that are open access should still be available. Maybe not the opposite though.

Reply:
It could be, but there are so many places where it depends on the vendor in a counterintuitive way that I think a vendor-independent algorithm like this one should be more conservative.